transport/http/dial.go

```diff
@@ -306,7 +306,7 @@ func addOpenTelemetryTransport(trans http.RoundTripper, settings *internal.DialS
 	if settings.TelemetryDisabled {
 		return trans
 	}
-	return otelhttp.NewTransport(trans)
+	return otelhttp.NewTransport(trans, settings.OpenTelemetryOptsHTTP...)
```

Comment:
For both this and the gRPC package this change is fine to make, but it won't actually enable the feature for most users -- that is because these packages, by default, delegate to our new auth library which has its own transport packages: https://pkg.go.dev/cloud.google.com/go/auth

At the very least I think we should add some TODOs and keep the issue open until this is plumbed all the way through to the new auth packages. See the `newClientNewAuth` function in this file as an example of how we delegate calls to this new library.

Reply:
Thank you @codyoss! I added a todo.